Blog: add interview with Anushka Saxena (#401)
cloudnative-pg/cloudnative-pg.github.io: content/blog/lfx-mentorship-documentation-anushka-saxena/index.md

---
title: "Updating CloudNativePG's documentation"
date: 2026-01-12
draft: false
image:
    url: anushka-saxena.jpg
    attribution:
authors:
 - fdrees
tags:
 - mentorship
 - lfx
 - cncf
 - PostgreSQL
 - Kubernetes
 - documentation
 - docusaurus
summary: "Meet the mentee: Anushka Saxena worked with the project maintainers on transforming 
the documentation for CloudNativePG, as part of the LFX mentorship program."
---

In the summer we wrote about how CloudNativePG was back for the September-

> October-November LFX term with [several projects for mentoring](https://cloudnative-pg.io/blog/2025-term3-lfx-cncf-mentorship/). One of them was around the Docs. 

Reviewer: "projects for mentoring" I think can be misread.
Maybe we could rephrase to "... CloudNativePG was back for the September-October-November term of LFX Mentorship with several projects.

> [Anushka Saxena](https://www.linkedin.com/in/-anushka-saxena/) is a Google Software Application Development Apprentice at Google, and is based in Bengaluru, India. She worked on improving the project’s documentation automation and multi-version support with mentors and CloudNativePG maintainers Gabriele Bartolini, Leonardo Cecchi, and Francesco Canovai. 

Reviewer 2: ```suggestion
[Anushka Saxena](https://www.linkedin.com/in/-anushka-saxena/) is a Software Application Development Apprentice at Google, and is based in Bengaluru, India. She worked on improving the project’s documentation automation and multi-version support with mentors and CloudNativePG maintainers Gabriele Bartolini, Leonardo Cecchi, and Francesco Canovai. 
```

I had a chat with Anushka about her work, and about how she got into Tech in the first place. 

## Start at the beginning

Anushka has always been motivated by curiosity and the desire to understand how systems work. "Over time, open source became a natural extension of that curiosity because it offered a way to learn by contributing, asking questions, and building things that real people use." What really drew her in was when she realized that great software often faces issues not because of poor engineering, but because it’s hard to understand or adopt. "That realization shaped my interest in documentation modernization and contributor experience as first-class technical problems."

> Anushka is a member of several Cloud-focused Communities, as a student community leader for leading cloud providers i.e., AWS, GCP and Azure) and the Cloud Native ecosystem at large, with a particular focus on developer-centric projects where infrastructure, documentation, and community intersect. She's been active in CNCF communities, open-source initiatives (OpenSearch, AsyncAPI), and DevOps tooling.

Reviewer 2: ```suggestion
Anushka is a contributor to the Cloud Native ecosystem, with a particular focus on developer-centric projects where infrastructure, documentation, and community intersect. With a background in community leadership for major cloud providers, she remains active in CNCF initiatives, open-source projects like AsyncAPI, and the broader Developer tooling landscape.
```

Reviewer 2: I made these suggestions because my previous community leadership was back in university so just tweaked the words so that it doesn't sound like I'm still in college

> "One community that’s especially close to my heart is Data on Kubernetes (DoK). I first got involved during my college days as a volunteer for DoK Community Days, which was my introduction to cloud-native data workloads and open-source communities by Bart Farell, a mentor and a very well-known personality in the CNCF space. Coming back to the CloudNativePG ecosystem as a mentee later on felt like a full-circle moment!"

Reviewer 3: ```suggestion
"One community that’s especially close to my heart is Data on Kubernetes (DoK). I first got involved during my college days as a volunteer for DoK Community Days, which was my introduction to cloud-native data workloads and open-source communities by [Bart Farell](uhttps://www.linkedin.com/in/bart-farrell/), a mentor and a very well-known personality in the CNCF space. Coming back to the CloudNativePG ecosystem as a mentee later on felt like a full-circle moment!"
```

She also worked on the Sidebar Position Multiplier, [a script](https://github.com/cloudnative-pg/docs/pull/22) that intelligently re-indexed our documentation pages (10, 20, 30...) to allow future contributors to insert new pages without breaking the order. Finally, she helped architect the [Auto-Sync Pipeline](https://github.com/cloudnative-pg/docs/pull/18), a GitHub Action that automatically pulls documentation from the main repo into the docs repo whenever a new release is published. This involved writing a reusable import script, integrating it into a GitHub Actions workflow triggered by upstream releases, and validating the output in a Docusaurus-based documentation setup. 

"A key part of my approach was treating documentation as production infrastructure: iterating locally, validating assumptions, opening pull requests, and refining solutions based on maintainer feedback. The goal was not just to make things work once, but to reduce long-term manual effort and make the documentation workflow more contributor-friendly and reliable."

> Anushka says that working with Gabriele, Francesco, and Leonardo was the highlight of her mentorship. "We had a very collaborative rhythm!" Through Slack, GitHub issues, weekly sync calls and pull requests, the team was responsive and generous with feedback, making it easy to iterate and improve. Anushka appreciated that discussions were clear, constructive, and focused on long-term project health rather than quick fixes. "Gabriele and Leonardo constantly provided the high-level vision that helped me see how my small scripts would affect the project as a whole."

Reviewer 2: ```suggestion
Anushka says that working with mentors Gabriele, Francesco, and Leonardo was the highlight of her mentorship. "We had a very collaborative rhythm!" Through Slack, GitHub issues, weekly sync calls and pull requests, the team was responsive and generous with feedback, making it easy to iterate and improve. Anushka appreciated that discussions were clear, constructive, and focused on long-term project health rather than quick fixes. "Gabriele, Francesco, and Leonardo constantly provided the overall high-level vision that helped me see how my small scripts would affect the project as a whole."
```

I had a chat with Anushka about her work, and about how she got into Tech in the first place. 

## Start at the beginning

Anushka has always been motivated by curiosity and the desire to understand how systems work. "Over time, open source became a natural extension of that curiosity because it offered a way to learn by contributing, asking questions, and building things that real people use." What really drew her in was when she realized that great software often faces issues not because of poor engineering, but because it’s hard to understand or adopt. "That realization shaped my interest in documentation and contributor experience as first-class technical problems."  

Anushka is a contributor to the Cloud Native ecosystem, with a particular focus on developer-centric projects where infrastructure, documentation, and community intersect. With a background in community leadership for major cloud providers, she remains active in CNCF initiatives, open-source projects like AsyncAPI, and the broader Developer tooling landscape.

> "One community that’s especially close to my heart is [Data on Kubernetes (DoK)](https://dok.community/). I first got involved during my college days as a volunteer for DoK Community Days, which was my introduction to cloud-native data workloads and open-source communities by [Bart Farell](uhttps://www.linkedin.com/in/bart-farrell/), a mentor and a very well-known personality in the CNCF space. Coming back to the CloudNativePG ecosystem as a mentee later on felt like a full-circle moment!"

Reviewer: ```suggestion
"One community that’s especially close to my heart is [Data on Kubernetes (DoK)](https://dok.community/). I first got involved during my college days as a volunteer for DoK Community Days, which was my introduction to cloud-native data workloads and open-source communities by [Bart Farell](https://www.linkedin.com/in/bart-farrell/), a mentor and a very well-known personality in the CNCF space. Coming back to the CloudNativePG ecosystem as a mentee later on felt like a full-circle moment!"
```

There's a typo in the URL for Bart's linkedin.
But also, IMO having several Linkedin profiles in a post makes it weird.
How about Bart's homepage instead https://bartfarrell.com ?
